File: app/Global.java
import com.avaje.ebean.Ebean;
import models.User;
import play.Application;
import play.GlobalSettings;
import play.Play;
import utils.DemoData;

public class Global extends GlobalSettings {

    @Override
    public void onStart(Application application) {

        //Ebean.getServer(null).getAdminLogging().setDebugGeneratedSql(true);
        
        // load the demo data in dev mode if no other data exists
        if (Play.isDev() && (User.find.all().size() == 0)) {
            DemoData.loadDemoData();
        }

        super.onStart(application);
    }

}

File: app/Global.scala
import models.Region
import play.api._
import plugins.MongoDB
import play.api.Play.current
import utils.DemoData

object Global extends GlobalSettings {
  
  override def onStart(app: Application) {

    implicit val context = MongoDB.context
    
    if (app.mode.equals(Mode.Dev) && (Region.sync.find.size == 0)) {
      DemoData.loadDemoData()
    }
    
  }
  
}